### PR TITLE
Forbid constructors with `ForbiddenMethodCall`

### LINKED ISSUE #5185: Forbid constructor calls
## Expected Behavior of the rule

Forbid the usage of a constructor like the same way we have `ForbiddenMethodCall`.

## Context
I'd like to prevent the usage of the `java.math.BigDecimal(double)` constructor. See https://stackoverflow.com/a/7186298 for explanation.

### PR DESCRIPTION
**Pull request by BraisGabin**
Fix #5185

---

I have no idea what's the real difference between `fqNameOrNull` and `fqNameSafe` but with the former all works as expected so 🤷

### CODE REVIEW

**Review by cortinico — APPROVED** [review]

**Comment by schalkms** [comment]
Interesting question...
Hmm, maybe you might want to ask this in the Kotlin Slack channel.

> I have no idea what's the real difference between fqNameOrNull and fqNameSafe but with the former all works as expected so 🤷

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt [hunk 1]
```diff
@@ -61,7 +61,8 @@ class ForbiddenMethodCall(config: Config = Config.empty) : Rule(config) {
             "(i.e. `java.time.LocalDate(java.time.Clock)`) which would report only call " +
             "with this concrete signature. If you want to forbid an extension function like" +
             "`fun String.hello(a: Int)` you should add the receiver parameter as the first parameter like this: " +
-            "`hello(kotlin.String, kotlin.Int)`"
+            "`hello(kotlin.String, kotlin.Int)`. To forbid constructor calls you need to define them with `<init>` " +
```
Comment (schalkms) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt:64]:
```suggestion
            "`hello(kotlin.String, kotlin.Int)`. To forbid constructor calls you need to define them with `<init>`, " +
```

**Review by schalkms — APPROVED** [review]

**Comment by schalkms** [comment]
Maybe, a hint should be given in the description of this rule that constructors are also supported.

**Comment by BraisGabin** [comment]
> Maybe, a hint should be given in the description of this rule that constructors are also supported.

Agree, what do you think about it now?

**Pull request merged**